openfasoc/generators/glayout/glayout/flow/blocks/elementary/transmission_gate/transmission_gate.py

```diff
@@ -0,0 +1,428 @@
+from typing import ClassVar, Optional, Any, Union, Literal, Iterable, TypedDict
+import math
+#from glayout.flow.pdk.gf180_mapped import gf180
+from glayout.flow.pdk.sky130_mapped import sky130_mapped_pdk as sky130
+from glayout.flow.pdk.mappedpdk import MappedPDK
+from glayout.flow.pdk.util.comp_utils import evaluate_bbox
+from glayout.flow.pdk.util.port_utils import set_port_orientation, rename_ports_by_orientation, create_private_ports
+from gdsfactory import Component
+from gdsfactory.components import rectangle
+from glayout.flow.primitives.fet import pmos
+from glayout.flow.primitives.fet import nmos
+from glayout.flow.routing.straight_route import straight_route
+from glayout.flow.routing.c_route import c_route
+from glayout.flow.routing.L_route import L_route
+from glayout.flow.routing.smart_route import smart_route
+from glayout.flow.placement.two_transistor_interdigitized import two_pfet_interdigitized, two_nfet_interdigitized, two_transistor_interdigitized
+from glayout.flow.placement.common_centroid_ab_ba import common_centroid_ab_ba
+from glayout.flow.pdk.util.comp_utils import prec_ref_center, movey, evaluate_bbox, align_comp_to_port
+from glayout.flow.primitives.via_gen import via_stack
+from gdsfactory.cell import cell
+from glayout.flow.spice import Netlist
+from glayout.flow.pdk.util.snap_to_grid import component_snap_to_grid
+from glayout.flow.primitives.guardring import tapring
+
+def transmission_gate_netlist(
+	pdk: MappedPDK,
+	width: float,
+	multipliers: int,
+	length: float,
+	subckt_only: Optional[bool] = False
+) -> Netlist:
+	if length is None:
+		length = pdk.get_grule('poly')['min_width']
+	if width is None:
+		width = 3
+	mtop = multipliers if subckt_only else 1
+	model_pmos = pdk.models['pfet']
+	model_nmos = pdk.models['nfet']
+	print(f"model_pmos: {model_pmos}")
+	source_netlist = """.subckt {circuit_name} {nodes} """ + f'l={length} w={width} m={mtop} ' + """
```

Comment:
You do not need to write the spice for the netlist. It can be constructed using the spice of the sub-devices (FETs). Refer to https://github.com/idea-fasoc/OpenFASOC/blob/e7f461123dd9e95216ac9cd59f057c8922fe47de/openfasoc/generators/glayout/glayout/flow/blocks/composite/opamp/opamp.py#L26